crates/languages/src/python.rs

```diff
@@ -791,11 +832,23 @@ impl ToolchainLister for PythonToolchainProvider {
                     });
 
             // Compare project paths against worktree root
-            let proj_ordering = || match (&lhs.project, &rhs.project) {
-                (Some(l), Some(r)) => (r == &wr).cmp(&(l == &wr)),
-                (Some(l), None) if l == &wr => Ordering::Less,
-                (None, Some(r)) if r == &wr => Ordering::Greater,
-                _ => Ordering::Equal,
+            let proj_ordering = || {
+                let lhs_project = lhs
+                    .project
+                    .as_ref()
+                    .or(get_venv_parent_dir(lhs).as_ref())
+                    .map(|p| p.to_path_buf());
+                let rhs_project = rhs
+                    .project
+                    .as_ref()
+                    .or(get_venv_parent_dir(rhs).as_ref())
+                    .map(|p| p.to_path_buf());
+                match (&lhs_project, &rhs_project) {
+                    (Some(l), Some(r)) => (r == &wr).cmp(&(l == &wr)),
```

Comment:
To be fair we should probably be checking whether the worktree root is a prefix of a candidate venv. That's obviously not a regression from your PR or anything, but maybe it'd be worthwhile to do that here?

Reply:
Would that check go in this match? Also, are you saying we shouldn't show any toolchains where the worktree root is not a prefix?

Reply:
That check would probably go in this match, yes.
I'm not saying we should not show venvs outside of current worktree - we should just prioritize ones for which the worktree root is a parent. We could even go an extra mile and sort based on distance from worktree root. 

Either way, maybe we could do that in separate PR on a second thought.

Reply:
Yep.